json-maker.py

```diff
@@ -116,65 +212,39 @@ def main():
     if modulesoverride == 'auto':
         modulesoverride = "overrides/modules-{}.json".format(gameinfo['name'])
 
-    if os.path.isfile(startoverride):
-        jsondata['modules'][0]['sources'].append(
-            collections.OrderedDict([
-                ("type", "file"),
-                ("path", startoverride)
-            ])
-        )
-    if os.path.isfile(configureoverride):
-        jsondata['modules'][0]['sources'].append(
-            collections.OrderedDict([
-                ("type", "file"),
-                ("path", configureoverride),
-                ("dest-filename", "configure")
-            ])
-        )
+    gamemodule = getGameModule(
+        args.gamemodule,
+        gameinfo,
+        args.installer,
+        startoverride,
+        configureoverride,
+        args.extra,
+    )
+    jsondata['modules'].insert(0, gamemodule)
+
     if os.path.isfile(modulesoverride):
         moduledata = "{}"
         with open(modulesoverride, 'r') as f:
-            moduledata = json.load(
-                            f, object_pairs_hook=collections.OrderedDict)
+            moduledata = json.load(f, object_pairs_hook=OrderedDict)
 
         for module in moduledata:
             jsondata['modules'].append(module)
 
-    for i, v in enumerate(args.extra):
-        jsondata['modules'][0]['sources'].append(
-            collections.OrderedDict([
-                ("type", "file"),
-                ("path", v),
-                ("dest-filename", "installer-{}.sh".format(i+1))
-            ])
-        )
-
-    if not gameinfo['arch']:
-        print(
-            "WARNING: Arch not specified, and not found in archlist.json - "
-            "defaulting to x86_64!")
-        gameinfo['arch'] = 'x86_64'
-    if gameinfo['arch'] == 'i386+x86_64':
-        gameinfo['arch'] = 'x86_64'
-
-    # app-id cannot start with a digit. Add an underscore if needed.
-    if gameinfo['name'][0].isdigit():
-        jsondata['app-id'] = 'com.gog._'+gameinfo['name']
-
-    outname = ("gen_com.gog.{}.json".format(gameinfo['name'])
+    outname = ("gen_{}.json".format(gameinfo['app-id'])
                if args.output == 'auto' else args.output)
 
     with open(outname, 'w') as outfile:
         json.dump(jsondata, outfile, indent=4)
 
+    buildarch = "x86_64" if gameinfo['arch'] == "i386" else gameinfo['arch']
```

Comment:
Since `gameinfo[arch]` is set to either i386 or x86_64 in `getGameInfo` (as long as the user didn't try something fun like `--arch arm64`), seems buildarch could just always be set to "x86_64".